src/js/controller/LayersListController.js

```diff
@@ -152,8 +189,10 @@
     } else if (action == 'merge') {
       this.mergeDownCurrentLayer_();
     } else if (action == 'edit') {
-      this.renameCurrentLayer_();
+      this.openCurrentLayerRenameForm_();
+      return;
```

Comment:
This can be confusing to use return as function "stopper", but you do it too so I guess you're ok with that.